cartographer/io/file_writer.cc

```diff
@@ -16,40 +16,61 @@
 
 #include "cartographer/io/file_writer.h"
 
+#include <fstream>
+#include "cartographer/common/make_unique.h"
+
 namespace cartographer {
 namespace io {
 
-StreamFileWriter::StreamFileWriter(const std::string& filename)
-    : filename_(filename), out_(filename, std::ios::out | std::ios::binary) {}
+StreamWriter::StreamWriter(std::unique_ptr<std::ostream> ostream,
+                           const std::string& filename)
+    : filename_(filename), is_closed_(false), out_(std::move(ostream)) {}
 
-StreamFileWriter::~StreamFileWriter() {}
+StreamWriter::~StreamWriter() {}
 
-bool StreamFileWriter::Write(const char* const data, const size_t len) {
-  if (out_.bad()) {
+bool StreamWriter::Write(const char* const data, const size_t len) {
+  if (out_->bad() || is_closed_) {
```

Comment:
it is allowed to write short statements in one line if they fit, i.e.:
`if (out_->bad() || is_closed_) return false;`